src/main.c

```diff
@@ -48,6 +48,8 @@ char __noinit __stack cdcacm_setup_stack_area[CDCACM_STACKSIZE];
 char __noinit __stack baudrate_reset_stack_area[RESET_STACKSIZE];
 char __noinit __stack usb_serial_stack_area[USBSERIAL_STACKSIZE];
 
+const char *firmware_version="CODK-M_1.6.0";
```

Comment:
OK ignore the BLE part. We want to keep a separate BLE firmware version number.

But, I was talking with Sidney, and I think we do want a bit more control over the location of this string.
So, I propose instead to declare an area for this version string in the linker script, at the very end of flash.

Then you can use the linker script symbols in C code to place the version string in the right area, and then the last N bytes of the binary file will be guaranteed to always contain that version string. So no grepping needs to happen, just read a fixed chunk from the very end of the binary file